include/seastar/coroutine/exception.hh

```diff
@@ -104,6 +111,11 @@ exception make_exception(T&& t) noexcept {
 /// ```
 /// co_await coroutine::return_exception(std::runtime_error("something failed miserably"));
 /// ```
+[[nodiscard]]
+inline exception return_exception_ptr(std::exception_ptr ex) noexcept {
```

Comment:
Doesn't this pass-by-value doesn't force the compiler to do an extra copy of the exception_ptr... Since exception_ptr has a non-trivial copy constructor (which needs to deal with the reference count), I don't see how the compiler can avoid generating it...
I would have done std::exception_ptr& and std::exception_ptr&& variants - not pass by value.

Reply:
ׁHmm, I think you're right. Ok. The extra move can (and should) be optimized out, so it's fine.

Reply:
That said, moving an exception_ptr should be cheap (same as a std::shared_ptr I think) as it doesn't copy the exception object.